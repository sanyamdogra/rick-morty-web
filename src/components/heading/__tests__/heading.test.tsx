import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import Heading from "..";

afterEach(cleanup);

test("should match the snapshot and render correct data", () => {
  const { asFragment } = render(
    <Heading primary="test-primary" secondary="test-secondary" />
  );

  expect(asFragment()).toMatchSnapshot();

  expect(screen.getByTestId("heading-primary")).toHaveTextContent(
    "test-primary"
  );

  expect(screen.getByTestId("heading-secondary")).toHaveTextContent(
    "test-secondary"
  );
});

test("should render correct data without secondary data", () => {
  render(<Heading primary="test-primary" />);

  expect(screen.getByTestId("heading-primary")).toHaveTextContent(
    "test-primary"
  );

  expect(screen.queryByTestId("heading-secondary")).toBeNull();
});
