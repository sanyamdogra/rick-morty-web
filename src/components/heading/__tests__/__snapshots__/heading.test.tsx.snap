// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot and render correct data 1`] = `
<DocumentFragment>
  <h1
    class="rick-morty-heading"
    data-testid="heading-primary"
  >
    test-primary
    <span
      class="rick-morty-heading__secondary"
      data-testid="heading-secondary"
    >
      test-secondary
    </span>
  </h1>
</DocumentFragment>
`;
